### PR TITLE
Merge deployment with develop

### PR DESCRIPTION
**Pull request by CVILLA09**
## Implemented Features
- **User Authentication**: Utilized the Devise gem to secure the app and manage user sessions. Users can now register, log in, and log out, with all data being private and secure.
- **Category Management**: Users can create, view, and delete categories. Each category showcases its name, icon, and the total amount spent.
- **Transaction Management**: Implemented the functionality for users to add and view transactions within categories, with a detailed view of the total amount spent.
- **UI Compliance**: Followed the given design guidelines to ensure the UI is consistent with specified colors, typography, and layout.
- **RESTful Routing**: Structured the app's routing to follow RESTful conventions, enhancing readability and maintainability of the code.

## Technical Details
- **Database**: Configured PostgreSQL as the production and development database.
- **Testing**: Created unit and integration tests for models and controllers to ensure reliability and prevent regressions.
- **Frontend**: Employed ERB for dynamic view templates, ensuring a seamless user experience.
- **Deployment**: The app is deployed to Render and is accessible online for review.

## Database Schema
Followed the ERD diagram to represent the app's data structure, ensuring that the database schema is optimized for efficiency and scalability.

## Code Quality
- Ensured no linter errors and followed best Ruby and Rails practices for clean and maintainable code.
- Implemented N+1 query solutions to optimize database interactions.

## Video Documentation
- Recorded a comprehensive video presenting the app's features, the design, and interesting pieces of code. ([Link](https://www.loom.com/share/9511907168644460af0de551c2e22049?sid=19e482e2-47fc-42db-8c63-97f8e2bdc144))

## Looking Forward
- Eager to receive feedback to improve and learn from the review process.
- Kindly review the changes and provide your valuable feedback.

### CODE REVIEW

**Comment by CVILLA09** [comment]
# Hello Reviewer!



## Help please! 

### Issue with Application Deployment on Render:

I'm encountering a problem with the deployment of my application on Render. Despite numerous attempts, I'm unable to resolve this issue. The site opens at 'https://budget-app-web-service.onrender.com/', displaying the splash screen with 'Log In' and 'Sign Up' options. However, clicking on either option results in the following error:

**"We're sorry, but something went wrong. If you are the application owner, check the logs for more information."**

I urgently request your assistance in resolving this matter, as I'm running out of time to get my project approved. I believe this is the only remaining issue, as the application seems to be functioning perfectly otherwise. Thank you in advance for your understanding and help.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @CVILLA09,






Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Nice project implementation.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -81,8 +81,8 @@
 
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-- [Live Demo Link]()
-- [Video presentation]()
+- [Live Demo Link](https://budget-app-web-service.onrender.com)
+- [Video presentation](https://www.loom.com/share/9511907168644460af0de551c2e22049?sid=19e482e2-47fc-42db-8c63-97f8e2bdc144)
```
[the remark this answers is not in the document]
Comment (CVILLA09) [reply README.md:85]: @zieeco 

Come on bro!! 

You didn't even helped me with the thing I was asking for help!

File: README.md [hunk 2]
```diff
@@ -81,8 +81,8 @@
...
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
[the remark this answers is not in the document]
Comment (CVILLA09) [reply README.md:88]: @zieeco 

What are you talking about?
Captura de pantalla 2023-12-02 a la(s) 5 18 24 p m

**Pull request merged**